EXT_BeamPhysics.md

```diff
@@ -0,0 +1,269 @@
+coordinate (typically called **s**) typically runs through the nominal centers of the elements
+in the machine. Typically, the **lattice** coordinate system is used to describe misalignments
+of lattice elements with respect to the rest of the lattice.
+
+- **Group**: A **group** is a container structure containing
+a set of zero or more **attributes**, a set of zero or more **groups** (which can be called
+**sub-groups**), and a set of zero or more **datasets**. Note: In HDF5, these are also
+called **groups**.
+
+- **Dataset**: A **dataset** is a structure that contains a set of zero or more **attributes** and a
+data aray (which may be multidimensional).  Note: In HDF5, these are also called **datasets**.
+
+- **Record**: A **record** is a **group** (without any sub-groups) or a **dataset** that contains data
+on a physical quantity like particle charge or electric field. There are two types of **records**:
+    - **scalar records** hold scalar quantity
+    values (like particle charge). If all the particles have the same charge, the value of the charge
+    is stored as an attribute of the **record** and there is no associated data array. That is, the
+    **record** is a **group**. If the particles have differing charges, the values are stored in an array
+    of the **scalar record**. In this case the **scalar record** is a **dataset**.
+    - **vector records** hold a set of **datasets**. In this case the **record** is a **group**.
+        - Example: A **record** named **E** for holding electric field values may have three datasets holding
+        the components of the field named **E/x**, **E/y**, and **E/z**.
+
+- **Attribute**: An **attribute** is a variable associated with a **group** along with a
+value. Example: **snapshotPath** is a string variable associated with the root **/** group.
+
+- **Polar coordinates**: **Polar** coordinates are **(r, theta, phi)** where **r** is the radius, **theta** is he angle
+from the **z** or **s** axis, and **phi** is the projected azimuthal angle in the **(x, y)**
+plane.
+
+- **Particle Root Group**: The **Particle Root Group** is the root group for specifying a group of particles. There can be multiple **particle root groups** in a data file. For example, a set of particle bunches might specify one particle root group for each bunch. See the Base Standard for more information.
+
+Notes:
+------
+
+- When using the **lattice** coordinate system, the `position` coordinates are **(x, y, s)** where
+nominally **x** is the "radial" component,
+**y** is the "vertical" coordinate, and **s** is the lattice longitudinal coordinate.
+
+
+Additional File Root Group (path `/`) Attributes
+-------------------------
+
+The following attributes are defined for the file root group.
+
+- `latticeName`
+  - Type: Optional *(string)*
+  - Description: The name of the lattice.
+
+- `latticeFile`
+  - Type: Optional *(string)*
+  - Description: The location of the root lattice file.
```

Comment:
The lattice file is not needed for, say, viewing the data so in that sense there is no **direct** dependency. However, if one wanted to propagate the particles further one would need an appropriate lattice file. 

If there were many attributes used here to store lattice information I would agree with you about trying to store too much. However this is only one attribute and there will be circumstances where this information will be very valuable so on the whole I would say the benefits outweigh the the cost of the extra storage.

Reply:
Ok, sorry I wasn't very clear ; when mentioning "trying to store too much information", the main disadvantage I had in mind was not the storage space (which, as you mentioned, is minimal), but rather the increase in complexity/robustness of the standard. 
In particular, here I am worried about robustness when moving the openPMD file (and its "attached" lattice file) on a different machine. Also: should the lattice file obey a particular standard?

Reply:
This does not impact the robustness of the Standard since data reader programs will not be affected. That is this does not affect the reading of all other parameters in the file. 

There will be cases where program A creates a particle data file which is handed off to program B but program B uses a different lattice input format so program B will be using a different lattice file in any case. The point is that storing the information as to the lattice file used with the simulation that calculated the particle positions can be useful information. Just like storing  the creation `date`, `machine` and other parameters as specified by the Standard.

Reply:
You know that modern file formats such as `HDF5` support references to groups/data sets in external files?

  https://support.hdfgroup.org/HDF5/doc/TechNotes/ExternalFiles.html

There is no need to reinvent this in openPMD on a higher level - it's fully [transparent](https://en.wikipedia.org/wiki/Transparency_(human%E2%80%93computer_interaction)) to the user / data reader.

For "file format backends" that do not support such references, the data can just be included.

Reply:
@ax3l: I don't understand how your comment relates to including a lattice file name. Can you elaborate?

Reply:
VC: does not apply here, the file is just an ASCII file

Reply:
@ax3l You lost me what does "VC" mean?

Reply:
What we discussed was: should we include the "lattice file", which can be constant for a whole simulation, in each file. I mentioned that HDF5 supports "external files" for this purpose, just like links, which avoid writing the same data over and over again. But then we realized that the lattice file is not an openPMD or HDF5 file but a plain ASCII file.

Reply:
>The question that remains: should we specify how a "lattice file" looks here? It's probably not clear to a reader (at least it is not to me) :)

Standardization of the lattice file is not possible. For example, the syntax of a lattice file for use by ASTRA is different from Bmad.